### PR TITLE
Add blade command services

### PR DESCRIPTION
**Pull request by lovettli**
Add new blade command services.
execute 'blade services' will output all of service extension points
execute 'blade services -s <service-name>' will output bundle name and version that contain this service

### CODE REVIEW

File: com.liferay.blade.cli/src/com/liferay/blade/cli/ServicesOptionAdapter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package com.liferay.blade.cli;
+
+import java.util.List;
+import java.util.Map;
+
+import aQute.lib.getopt.CommandLine;
+
+public class ServicesOptionAdapter implements ServicesOptions{
```
Comment (lovettli) [inline com.liferay.blade.cli/src/com/liferay/blade/cli/ServicesOptionAdapter.java]: This Adapter is able to help developer to get bundle name and version from our code. please look at follow test class ServicesTest.

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
Lets add this code to our IDE directly, I don't think this will be useful to command line users outside of the use-case of the IDE wizards.

**Comment by lovettli** [comment]
Hey Greg,
If put this code in our IDE, That's mean if user want to use command create service project or service wrapper ,we don't know this service belong to which bundle. So we can't be able to configure dependencies. Compiling will be failed.
this function "public String[] parseSymbolicName(String info)" will provide to search service bundle and version , it can be used by blade create command to perfect the dependency of build.gradle file.
So what do you think ?

**Comment by gamerson** [comment]
So the same code used to connect to portal can be called from IDE just as
it could from blade.  So its exactly the same.  I just don't think this
function is useful outside of IDE so I want it to go in IDE instead.

On Tue, Dec 29, 2015 at 8:54 PM, Lovett Li <notifications@github.com> wrote:

> Hey Greg,
> If put this code in our IDE, That's mean if user want to use command
> create service project or service wrapper ,we don't know this service
> belong to which bundle. So we can't be able to configure dependencies.
> Compiling will be failed.
> this function "public String[] parseSymbolicName(String info)" will
> provide to search service bundle and version , it can be used by blade
> create command to perfect the dependency of build.gradle file.
> So what do you think ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gamerson/liferay-blade-tools/pull/49#issuecomment-167922728>
> .
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

**Comment by lovettli** [comment]
OK. you mean we don't worry about the dependency of build.gradle file is right or wrong when user create a service project via blade. Right? we will fix dependency problem in our IDE side Right?

**Comment by gamerson** [comment]
Right.

On Tue, Dec 29, 2015 at 9:37 PM, Lovett Li <notifications@github.com> wrote:

> OK. you mean we don't worry about the dependency of build.gradle file is
> right or wrong when user create a service project via blade. Right? we will
> fix dependency problem in our IDE side Right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gamerson/liferay-blade-tools/pull/49#issuecomment-167931430>
> .
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

**Comment by lovettli** [comment]
OK.got it.Thanks

**Pull request closed without merging**